retail/interactive-tutorial/RetailEvents.Samples/ImportUserEventsGcsSample.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_import_user_events_from_gcs]
+// Import user events into a catalog from GCS using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using System;
+
+/// <summary>
+/// The import user events gcs sample class.
+/// </summary>
+public class ImportUserEventsGcsSample
+{
+    private const string GcsEventsObject = "user_events.json";
+
+    // To check error handling use the json with invalid user event:
+    // private const string GcsEventsObject = "user_events_some_invalid.json";
+
+    /// <summary>
+    /// Get import user events from gcs request.
+    /// </summary>
+    /// <param name="gcsObjectName">The name of the Gcs object.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>The import user events request.</returns>
+    private static ImportUserEventsRequest GetImportUserEventsGcsRequest(string gcsObjectName, string projectId, string bucketName)
+    {
+        string eventsBucketName = bucketName ?? Environment.GetEnvironmentVariable("EVENTS_BUCKET_NAME");
+        string gcsBucket = $"gs://{eventsBucketName}";
+        string gcsErrorsBucket = $"{gcsBucket}/error";
```

Comment:
This is not a bucket
```suggestion
        string gcsErrorsPrefix = $"{gcsBucket}/error";
```

Reply:
done